app/main.py

```diff
@@ -1,2 +1,117 @@
+from typing import Any, List, Optional, Iterator
+
+
+class Node:
+    """Node to store key-value pairs in the hash table."""
+    def __init__(self, key: Any, value: Any) -> None:
+        self.key: Any = key
+        self.value: Any = value
+        self.hash: int = hash(key)
+        self.next: Optional[Node] = None
```

Comment:
The use of `Optional[Node]` in the type hint for `self.next` assumes that `Node` is already defined. In some Python environments, this might cause issues. Consider using a forward reference by quoting 'Node' as 'Optional["Node"]'.